src/panels/config/script/ha-script-field-row.ts

```diff
@@ -269,6 +269,14 @@ export default class HaScriptFieldRow extends LitElement {
     this._errorKey = undefined;
     this._uiError = undefined;
 
+    // If we render the default with an incompatible selector, it risks throwing an exception and not rendering.
+    // Clear the default when changing the selector type.
+    if (
+      Object.keys(this.field.selector)[0] !== Object.keys(value.selector)[0]
+    ) {
+      delete value.default;
+    }
+
```

Comment:
Had to add this back in as I think it's going to be very difficult to sanitize the entire world of possible inputs to every selector. I did try to fix a couple selectors that would break when you changed their options, so for now this only clears when you change type.